src/Main.cpp

```diff
@@ -225,10 +231,15 @@ class SoftwareInterlocTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE>
     // Create the object so it can register it's callbacks
     // TODO: remove once it is used somewhere else
     IInterloc& m_interloc;
+    NotificationQueue<MessageDTO>& m_interlocMessageQueue;
     IInterlocMessageHandler& m_interlocMessageHandler;
 
     void task() override {
         while (true) {
+            // Verify that we have a message to process
+            if (m_interlocMessageQueue.isEmpty()) {
+                m_interlocMessageQueue.wait(UINT32_MAX);
```

Comment:
Do we really want to be blocking? We could make the delay relatively long, but still have a timeout in case the notifications fails for some reason. If we keep it blocking, we should use portMAX_DELAY. For posix, I think we have a define for it in propolis, and for freeRTOS it would use the one for the port and would be safer

Reply:
I'm not sure why we wouldn't want to block, we added that to  avoid useless processing. This removes the need for it. The notification shouldn't fail, if they do, we have bigger underlying problems with the design/code. portMAX_DELAY is the same thing as UINT32_MAX since all our builds are at least 32 bits.

Reply:
I agree with @g-lesssard on this. Checking if the queue has something in it every few ms won't impact performance at all while possibly preventing bugs with edge cases we haven't thought of.

Reply:
Changed it to 500 ms. though generally "partially hiding" bugs aren't the way to go  since it becomes hard to reproduce bugs and can hide bigger inner problems in the code.